### PR TITLE
[MAINT] relocate functions from nilearn.datasets.utils

### LINKED ISSUE #4139: Relocate functions from datasets.utils
Suggestion to do in a separate PR:
What do you think about deprecating this file and relocating these two functions? `load_sample_motor_activation_image` could go under `func.py`; not so sure yet about `get_data_dirs`. Or alternatively renaming `utils.py`
Not sure if it's worth it but something to think about if any confusion arises from having `_utils.py` and `utils.py`

_Originally posted by @ymzayek in https://github.com/nilearn/nilearn/pull/4086#discussion_r1384846240_

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4139

Changes proposed in this pull request:

- relocate load_sample_motor_activation_image
- throw deprecation warning when imported from the older location

## TODO

- [x] add test for deprecation warning
- [ ] relocate `get_data_dirs`

### CODE REVIEW

**Comment by Remi-Gau** [comment]
regarding https://github.com/nilearn/nilearn/blob/af3745636152fe55e128a936e671e3a58536fe02/nilearn/datasets/utils.py#L9

If we still want to move it, maybe actually adding directly to 
`nilearn/datasets/__init__.py` ? Not sure how pythonic that is.

**Review by bthirion — COMMENTED** [review]
I guess you need to test the warnings but this looks good so far.

File: nilearn/datasets/utils.py [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
-from pathlib import Path
+from warnings import warn
 
 from .._utils import fill_doc
 
+_GENERAL_MESSAGE = (
+    "The import path 'nilearn.utils' will deprecated in version 0.13. "
+    "Importing from 'nilearn.utils' will be possible "
+    "at least until release 0.13.0."
+)
```
Comment (ymzayek) [inline nilearn/datasets/utils.py:11]: that should reference 'nilearn.datasets.utils' no?

Comment (Remi-Gau) [reply nilearn/datasets/utils.py:11]: indeed. 😅

**Comment by Remi-Gau** [comment]
Don't think it is worth relocating `get_data_dirs` unless we get active complaint from users (or our future selves) that having `nilearn.datasets._utils` and `nilearn.datasets.utils` is confusing.

**Review by ymzayek — APPROVED** [review]
Good with me

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Pull request merged**